lib/commands/xctest.js

```diff
@@ -0,0 +1,209 @@
+
+    // One single string, just return the string
+    if (lines.length === 1 && !lines[0].includes('|')) {
+      return lines[0];
+    }
+
+    const results = [];
+    for (const line of lines) {
+      // The properties are split up by pipes and each property
+      // has the format "Some Key : Some Value"
+      const properties = line.split('|');
+
+      // Parse each property
+      const output = {};
+      let entryIndex = 0;
+      for (let prop of properties) {
+        if (entryIndex === 0) {
+          // The first property only contains one string that contains
+          // the test name (e.g.: 'XCTesterAppUITests - XCTesterAppUITests.XCTesterAppUITests/testExample')
+          output.testName = prop.trim();
+        } else {
+          let [key, value] = prop.split(':');
+          output[parseKey(key.trim())] = parseValue(value.trim());
+        }
+        entryIndex++;
+      }
+      // Add this line to the results
+      results.push(output);
+    }
+    return results;
+  }
+}
+
+/**
+ * @typedef {Object} RunXCUITestResponse
+ *
+ * @property {Array<Object>} results The results of all the tests with information (passed, crashed, duration, etc...)
+ * @property {number} code The exit code of the process
+ * @property {string} signal The signal that terminated the process (or null)
+ *
+ */
+
+/**
+ * @typedef {Object} RunXCUITestOptions
+ *
+ * @property {string!} testRunnerBundleId Test app bundle
+ * @property {string!} appUnderTestBundleId App-under-test bundle
+ * @property {string!} xcTestBundleID xctest bundle
+ * @property {string} testType [ui] XC test type. 'app', 'ui', or 'logic'
+ * @property {object} env Environment variables passed to test
+ * @property {Array<String>} args Test arguments
+ * @property {number} timeout [360000] Timeout if session doesn't complete after given time
+ * @returns {RunXCUITestResponse}
+ */
+
+/**
+ * Run an XCTest. Launches a subprocess that runs the XC test and blocks
+ * until it is complete.
+ *
+ * See https://fbidb.io/docs/test_execution for reference
+ *
+ * @param {RunXCUITestOptions} runXCUITestOptions
+ */
+commands.mobileRunXCTest = async function runXCTest ({
+  testRunnerBundleId,
+  appUnderTestBundleId,
+  xctestBundleId,
+  testType = 'ui',
+  env,
+  args,
+  timeout = XCTEST_TIMEOUT,
+}) {
+  const subproc = await assertIDB(this.opts).runXCUITest(
+        testRunnerBundleId, appUnderTestBundleId, xctestBundleId, {env, args, testType},
+  );
+  return await new B((resolve, reject) => {
+    let mostRecentLogObject = null;
+    let xctestTimeout;
+    if (timeout > 0) {
+      xctestTimeout = setTimeout(
+        () => reject(`Timed out after '${timeout}s' waiting for XCTest to complete`),
+        timeout
+      );
+    }
+
+    subproc.on('output', (stdout, stderr) => {
+      if (stdout) {
+        try {
+          mostRecentLogObject = parseXCTestStdout(stdout);
+        } catch (e) {
+          // Fails silently if log parsing fails.
```

Comment:
I'd rather print a debug line with a stack trace  here, so we know what to update